spec/Provider/PaymentTokenProviderSpec.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+
+declare(strict_types=1);
+
+namespace spec\BitBag\SyliusImojePlugin\Provider;
+
+use BitBag\SyliusImojePlugin\Provider\PaymentTokenProvider;
+use Doctrine\Common\Collections\ArrayCollection;
+use PhpSpec\ObjectBehavior;
+use Sylius\Bundle\PayumBundle\Model\PaymentSecurityTokenInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Payment\Model\PaymentInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class PaymentTokenProviderSpec extends ObjectBehavior
+{
+    public function let(
+        RepositoryInterface $orderRepository,
+        RepositoryInterface $paymentTokenRepository,
+    ): void {
+        $this->beConstructedWith($orderRepository, $paymentTokenRepository);
+    }
+
+    public function it_is_initializable(): void
+    {
+        $this->shouldHaveType(PaymentTokenProvider::class);
+    }
+
+    public function it_should_return_token_correctly(
+        Request $request,
+        OrderInterface $order,
+        PaymentSecurityTokenInterface $token,
+        PaymentInterface $payment,
+        RepositoryInterface $orderRepository,
+        RepositoryInterface $paymentTokenRepository,
+    ): void {
+        $orderNumber = 500;
+        $tokenHash = '3423423453fsxzc';
+
+        $transactionData = [
+            'transaction' => [
+                'orderId' => $orderNumber,
+                'tokenHash' => $tokenHash,
+            ]
+        ];
+
+        $request->getContent()
+            ->willReturn(json_encode($transactionData));
+
+        $orderRepository->findOneBy(['number' => $orderNumber])
+            ->willReturn($order);
+
+        $order->getPayments()
+            ->willReturn(new ArrayCollection([
+                $payment->getWrappedObject()
+            ]));
+
+        $payment->getState()
+            ->willReturn(PaymentInterface::STATE_NEW);
+
+        $payment->getDetails()
+            ->shouldBeCalled()
+            ->willReturn(['tokenHash' => $tokenHash]);
```

Comment:
```suggestion
        $payment->getDetails()
            ->shouldBeCalled()
            ->willReturn(['tokenHash' => $tokenHash]);
```

Please do not to mix stubs with mocks. Could you review the entire PR for this?